addons/mail/models/mail_channel.py

```diff
@@ -836,6 +836,10 @@ def channel_info(self, extra_info=False):
                 partner_channel = channel_partners.filtered(lambda pc: pc.partner_id.id == self.env.user.partner_id.id)
                 if partner_channel:
                     partner_channel = partner_channel[0]
+                    info['members'] = [{
+                        'id': partner_channel.id,
+                        'partner_id': partner_infos[partner_channel.partner_id.id],
+                    }]
```

Comment:
Adding all members was previously protected inside a `if channel.channel_type != 'channel':` condition.

This condition is critical to avoid having init_messaging loading tons of data and having an RPC returning several MB of data.

As we briefly discussed the other day, `channel_info` should take a param to control whether members have to be returned or not, because this can become very costly for channels with all users inside them.

The member list should be lazy-loaded for those channels.

Reply:
This line is tied to the comment https://github.com/odoo/odoo/pull/72170#discussion_r651895917

We used to add `currentPartner` to the thread if there was no members, but since members aren't partners anymore, I need to get at least the member that represents the current partner. So as a default case I add only one member, not all of them.

Reply:
Ok I get it, that sounds reasonable, but explain this as comment in the python then.

Reply:
Note that if we start doing that, we need to change the syntax. `members` key is assumed to be the full member list. The JS needs to know this is partial information in that case.

Reply:
The member list was already partial back when the `currentPartner` was added to the empty member list*,  is there a case in particular you have in mind? 


* https://github.com/odoo/odoo/pull/72170/files#diff-3f2d7f866710abf0f185f1dc327deaa1b5e893e68ef8e52df38e173cc06c544dL457